mailpoet/lib/Config/Initializer.php

```diff
@@ -319,6 +327,22 @@ public function initialize() {
     define(self::INITIALIZED, true);
   }
 
+  /**
+   * Walk around for getting this to work correctly
+   *
+   * Read more here: https://developer.wordpress.org/reference/functions/register_activation_hook/
+   * and https://github.com/mailpoet/mailpoet/pull/4620#discussion_r1058210174
+   * @return void
+   */
+  public function afterPluginActivation() {
+    if (!$this->wpFunctions->isAdmin() || !defined(self::INITIALIZED) || !$this->wpFunctions->getOption(self::PLUGIN_ACTIVATED)) return;
+
+    $this->changelog->redirectToLandingPage();
+
+    // done with afterPluginActivation actions
+    $this->wpFunctions->deleteOption(self::PLUGIN_ACTIVATED);
```

Comment:
@triple0t the method `redirectToLandingPage` is exiting the code earlier and this method - `deleteOption` - isn't getting reached and the option persists.
This behavior makes the redirect to "/wp-admin/admin.php?page=mailpoet-landingpage" occurs whenever any plugin is activated.
I've opened an issue: https://github.com/mailpoet/mailpoet/issues/5014

Reply:
Fixed in https://github.com/mailpoet/mailpoet/pull/5020
Thank you, @fseabra47am